Replace Appveyor by GitHub Actions for Windows

yaminijha1 opened the pull request:
---

<!--
Remove the fields that are not appropriate
Please include:
-->

**What's this PR do?**
 Fixes #1070  

This PR remove the Appveyor CI workflow and adds GitHub Actions on Windows build

yaminijha1 committed:
Add Github Action for windows build

Fixes #1070

This commit replaces Appveyor with a GitHub Actions workflow for Windows.
It uses `setup-node@v4` with `node-version: lts/*` and `checkout@v4`.
Also removes `appveyor.yml`.
With no conflict with current branch.

shubham-padia commented on .github/workflows/node.js.yml:
```diff
@@ -7,9 +7,32 @@ on:
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "17"
```
Can you change this to `lts/*`, so we get the latest node version.

shubham-padia commented on .github/workflows/node.js.yml:
```diff
@@ -7,9 +7,32 @@ on:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
```
We should use setup-node@v4 since that's the latest version

shubham-padia reviewed (changes requested):
Sorry for the delay. Overall looks good, requested some changes. @yaminijha1 let me know if you will be interested to make those changes or not. If not, I can make those changes and add you as a co-author.

shubham-padia commented:
@yaminijha1 Can you squash the second commit into the first? Thanks!

yaminijha1 commented:
Hi ~shubham-padia,
This PR #1350 now fixes #1070 and incorporates all your suggestions:

Updated the Node.js version to lts/*
Switched to setup-node@v4
Squashed the commits into one, as requested

Please let me know if any further changes are needed.

shubham-padia commented:
@timabbott this should be ready for integration review

yaminijha1 committed:
Update node.js.yml

yaminijha1 committed:
Update node.js.yml

yaminijha1 commented:
@zulipbot @timabbott  this PR #1350  now fixes #1070 without any conflict without any conflict.

andersk commented:
- If we’re running more or less the same steps on both platforms, this should use a [matrix](https://docs.github.com/en/actions/how-tos/write-workflows/choose-what-workflows-do/run-job-variations).

- The extra `node --version` and `npm --version` are not needed because GitHub already includes these in the log under **Run actions/setup-node@v4** → **Environment details**. `python --version` is not needed because we do not use Python.

- The Appveyor badge needs to be removed from `README.md`.
